lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4064,12 +4066,12 @@ static SPIRVWord getBuiltinIdForIntrinsic(Intrinsic::ID IID) {
     return OpenCLLIB::Log10;
   case Intrinsic::log2:
     return OpenCLLIB::Log2;
+  case Intrinsic::maximumnum:
   case Intrinsic::maximum:
-    return OpenCLLIB::Fmax;
   case Intrinsic::maxnum:
     return OpenCLLIB::Fmax;
+  case Intrinsic::minimumnum:
   case Intrinsic::minimum:
```

Comment:
Revisiting this, it's actually incorrect to map `minimum` to `OpenCLLIB::Fmin` due to differences in NaN behaviour:

`llvm.minimum`:
> If either operand is a NaN, returns NaN.

`fmin`:
> If one operand is a NaN, fmin returns the other argument. If both arguments are NaNs, fmin returns a NaN. 

I've raised #3282 for this.

`llvm.minimumnum` follows `fmin`'s NaN behaviour, so this change looks good to me.

Reply:
> Revisiting this, it's actually incorrect to map `minimum` to `OpenCLLIB::Fmin` due to differences in NaN behaviour:

yes, both `minimum` and `maxnum` have the difference in NaN behaviour compared to OpenCL fmax/fmin